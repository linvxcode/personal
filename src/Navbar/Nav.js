import React, { useReducer } from "react";

import {
  Navbar,
  NavbarBrand,
  NavbarContent,
  NavbarItem,
  NavbarMenuToggle,
} from "@nextui-org/react";
import DarkMode from "../Component/DarkMode";
import TopNav from "../Component/TopNav";
import Btn from "../Component/Btn";

export default function App() {
  const [isMenuOpen, setIsMenuOpen] = useReducer((current) => !current, false);


  return (
    <Navbar
      maxWidth="full"
      className="fixed h-[80px] max-sm:px-0 px-12"
      isMenuOpen={isMenuOpen} onMenuOpenChange={setIsMenuOpen}
      
    >
      <NavbarContent >
        <NavbarMenuToggle
        
          aria-label={isMenuOpen ? "Close menu" : "Open menu"}
          className="lg:hidden"
        />
        <NavbarBrand>
          <Btn to="home" variant="light" className="px-0 font-bold text-inherit font-mons uppercase">
            M Fahlevi 
          </Btn>
        </NavbarBrand>
      </NavbarContent>
      <NavbarContent justify="end">
        <NavbarItem className="hidden lg:flex">
          <Btn to="about" color="light">
            About
          </Btn>
          <Btn to="skill" color="light">
            Technology
          </Btn>
          <Btn to="triple" color="light">
            Portofolio
          </Btn>
        </NavbarItem>
        <NavbarItem>
          <DarkMode />
        </NavbarItem>
      </NavbarContent>
      <div>
        <TopNav isOpen={isMenuOpen}  setIsMenuOpen={setIsMenuOpen} />
      </div>
    </Navbar>
  );
}
