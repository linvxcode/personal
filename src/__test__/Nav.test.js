

import React from "react";
import { render, screen } from "@testing-library/react";
import Nav from "../Navbar/Nav";

describe('Nav', () => {
    it('Render Nav', () => {
        const {Navbar} = render(<Nav />);
        expect(Navbar).toMatchSnapshot();
        const text = screen.getByText('M Fahlevi');
        expect(text).toBeInTheDocument();
    })
})