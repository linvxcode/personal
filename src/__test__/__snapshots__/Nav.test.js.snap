// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Render Nav 1`] = `undefined`;
